src/VideoCommand.cc

```diff
@@ -350,3 +360,194 @@ Json::Value FindVideo::construct_responses(
+            if (type == "interval") {
+                op_index = i;
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+int FindFrames::construct_protobuf(
+    PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    // We try to catch the missing attribute error before
+    // initiating a PMGD query
+    Json::ArrayIndex tmp;
+    bool is_interval = get_interval_index(cmd, tmp);
+    bool is_frames   = cmd.isMember("frames");
+
+    if (!(is_frames != is_interval)) {
+        error["status"]  = RSCommand::Error;
+        error["info"] = "Either one of 'frames' or 'operations::interval' "
+                        "must be specified";
+        return -1;
+    }
+
+    Json::Value results = get_value<Json::Value>(cmd, "results");
+    results["list"].append(VDMS_VID_PATH_PROP);
+
+    query.QueryNode(
+            get_value<int>(cmd, "_ref", -1),
+            VDMS_VID_TAG,
+            cmd["link"],
+            cmd["constraints"],
+            results,
+            get_value<bool>(cmd, "unique", false)
+            );
+
+    return 0;
+}
+
+Json::Value FindFrames::construct_responses(
+    Json::Value& responses,
+    const Json::Value& json,
+    protobufs::queryMessage &query_res,
+    const std::string &blob)
+{
+    const Json::Value& cmd = json[_cmd_name];
+
+    Json::Value ret = check_responses(responses);
+
+    auto error = [&](Json::Value& res)
+    {
+        ret[_cmd_name] = res;
+        return ret;
+    };
+
+    Json::Value& FindFrames = responses[0];
+
+    bool flag_empty = true;
+
+    for (auto& ent : FindFrames["entities"]) {
+
+        if(!ent.isMember(VDMS_VID_PATH_PROP)){
+            continue;
+        }
+
+        std::string video_path = ent[VDMS_VID_PATH_PROP].asString();
+        ent.removeMember(VDMS_VID_PATH_PROP);
+
+        if (ent.getMemberNames().size() > 0) {
+            flag_empty = false;
+        }
+
+        try {
+            std::vector<unsigned> frames;
+
+            // Copy of operations is needed, as we pass the operations to
+            // the enqueue_operations() method of ImageCommands class, and
+            // it should not include 'interval' operation.
+            Json::Value operations  = cmd["operations"];
+
+            Json::ArrayIndex interval_idx;
+            bool is_interval = get_interval_index(cmd, interval_idx);
+            bool is_frames   = cmd.isMember("frames");
+
+            if (is_frames) {
+                for (auto& fr : cmd["frames"]) {
+                    frames.push_back(fr.asUInt());
+                }
+            }
+            else if (is_interval) {
+                Json::Value interval_op = operations[interval_idx];
+                int i;
+                for (
+                    i =  get_value<int>(interval_op, "start");
+                    i <  get_value<int>(interval_op, "stop");
```

Comment:
I think it might make sense to read the JSON variables before the for loop. Philip has made me less trusting of the compiler over the years :)